blueprints/model/index.js

```diff
@@ -47,46 +48,96 @@ module.exports = {
       let attr;
       if (/has-many/.test(dasherizedType)) {
         let camelizedNamePlural = inflection.pluralize(camelizedName);
-        attr = dsAttr(dasherizedForeignModelSingular, dasherizedType);
-        attrs.push(camelizedNamePlural + ': ' + attr);
+        attr = {
+          name: dasherizedForeignModelSingular,
+          type: dasherizedType,
+          propertyName: camelizedNamePlural,
+        };
       } else if (/belongs-to/.test(dasherizedType)) {
-        attr = dsAttr(dasherizedForeignModel, dasherizedType);
-        attrs.push(camelizedName + ': ' + attr);
+        attr = {
+          name: dasherizedForeignModel,
+          type: dasherizedType,
+          propertyName: camelizedName,
+        };
       } else {
-        attr = dsAttr(dasherizedName, dasherizedType);
-        attrs.push(camelizedName + ': ' + attr);
+        attr = {
+          name: dasherizedName,
+          type: dasherizedType,
+          propertyName: camelizedName,
+        };
       }
+      attrs.push(attr);
 
       if (/has-many|belongs-to/.test(dasherizedType)) {
         needs.push("'model:" + dasherizedForeignModelSingular + "'");
       }
     }
 
+    let attrTransformer, attrSeparator;
+    if (process.env.EMBER_VERSION === 'OCTANE') {
```

Comment:
We could create a util function under `lib/utilities` to detect `OCTANE`.  This function could be in the same module than `lib/utilities/module-unification` and this one could be renamed it to `env-detector`.

I would prefer to do this change in a follow-up PR and based on the reviewer instructions.